packages/malloy-render/src/component/render-result-metadata.ts

```diff
@@ -21,283 +22,1562 @@
+          }))
+        );
+      }
+      current = current.parent;
+    }
+    return result;
+  }
+
+  getDrillExpressions(): string[] {
+    const drillValues = this.getDrillValues();
+    return drillValues.map(({field, value}) => {
+      const valueStr = valueToMalloy(value);
+      return `${field.drillExpression()} = ${valueStr}`;
+    });
+  }
+
+  getDrillEntries(): DrillEntry[] {
+    const drillValues = this.getDrillValues();
+    const result: DrillEntry[] = [];
+    for (const drill of drillValues) {
+      if (
+        drill.value.isNull() ||
+        drill.value.isTime() ||
+        drill.value.isString() ||
+        drill.value.isNumber() ||
+        drill.value.isBoolean()
+      ) {
+        result.push({field: drill.field, value: drill.value.value});
+      }
+    }
+    return result;
+  }
+
+  getDrillQuery(): string {
+    const whereClause = this.getDrillExpressions()
+      .map(entry => `    ${entry}`)
+      .join(',\n');
+    return `
+      run: ${this.field.root().sourceName} -> {
+  where:
+${whereClause}
+} + { select: * }`.trim();
+  }
+}
+
+export class ArrayCell extends CellBase {
+  public readonly values: Cell[] = [];
+  constructor(
+    public readonly cell: Malloy.CellWithArrayCell,
+    public readonly field: ArrayField,
+    public readonly parent: NestCell | undefined
+  ) {
+    super(cell, field, parent);
+    for (const value of this.cell.array_value) {
+      this.values.push(Cell.from(value, field.eachField, this));
+    }
+  }
+
+  get value() {
+    return this.values;
+  }
+}
+
+export class RepeatedRecordCell extends ArrayCell {
+  public readonly rows: RecordCell[];
+  public readonly fieldValueSets: Map<string, Set<CellValue>> = new Map();
+  constructor(
+    public readonly cell: Malloy.CellWithArrayCell,
+    public readonly field: RepeatedRecordField,
+    public readonly parent: NestCell | undefined
+  ) {
+    super(cell, field, parent);
+    this.rows = this.values as RecordCell[];
+    // First, create cells for all the rows
+    for (const row of this.rows) {
+      for (const column of row.columns) {
+        const field = column.field;
+        let valueSet = this.fieldValueSets.get(field.name);
+        if (valueSet === undefined) {
+          valueSet = new Set();
+          this.fieldValueSets.set(field.name, valueSet);
+        }
+        valueSet.add(column.value);
+      }
+    }
+    for (const [field, set] of this.fieldValueSets.entries()) {
+      this.field.registerValueSetSize(field, set.size);
+    }
+  }
+
+  get value() {
+    return this.rows;
+  }
+
+  // TODO I don't think it makes sense to override cellAtPath here -- what would it mean to get a particular column
+  // in a repeated record -- would you get the first row always??
+}
+
+export class RootCell extends RepeatedRecordCell {
+  constructor(
+    public readonly cell: Malloy.CellWithArrayCell,
+    public readonly field: RootField
+  ) {
+    super(cell, field, undefined);
+  }
+}
```

Comment:
I sorta forgot that I essentially just coerce the root field into a repeated record regardless... It's kinda fine?